classes/01class/exercises/c01-aws01/afaryy/README.md

```diff
@@ -0,0 +1,260 @@
+﻿# C01-AWS01
+
+## Commands Execution Output
+### 1. Use the CLI to create a new EC2 Instance inside the default VPC with a public IP address and connect to it though SSH.
+
+- Create a user and add aws configure profile
+```bash
+aws configure --profile devopsacademyau
+AWS Access Key ID [None]: ********
+AWS Secret Access Key [None]:********
+Default region name [None]: ap-southeast-2
+Default output format [None]:
+```
+- Get default VPC id and select one Subnetid:
+```bash
+aws ec2 describe-vpcs --filters "Name=isDefault, Values=true" |grep VpcId 
+# "VpcId": "vpc-c47d41a3",
+aws ec2 describe-subnets --filters "Name=vpc-id,Values=vpc-c47d41a3" |grep SubnetId
+#             "SubnetId": "subnet-08bf2850",
+#             "SubnetId": "subnet-d1e3f598",
+#             "SubnetId": "subnet-35b05453",
+
+```
+- Creating a Key Pair For the first EC2 instance:
+```
+aws ec2 create-key-pair --key-name JupmBoxKeyForFirstEC2Instance --query 'KeyMaterial' --output text > JupmBoxKeyForFirstEC2Instance.pem
+chmod 400 JupmBoxKeyForFirstEC2Instance.pem
+aws ec2 describe-key-pairs --key-nam JupmBoxKeyForFirstEC2Instance
+```
+- Creating a Security Group for JupmBoxKeyForFirstEC2Instance:
+```bash
+aws ec2 create-security-group --group-name sg_JupmBox --description "JumpBox Security Group" --vpc-id vpc-c47d41a3
+# "GroupId": "sg-06ae9a4efdd9232b8"
+```
+- ADD Security Group rule for allow to SSH connection for JupmBox:
+```bash
+# uese your own ip/32 instead of 0.0.0.0/0
+aws ec2 authorize-security-group-ingress --group-id sg-06ae9a4efdd9232b8 --protocol tcp --port 22 --cidr 0.0.0.0/0 
+```
+- Creat the first EC2 instance:
+https://docs.aws.amazon.com/cli/latest/reference/ec2/run-instances.html
+```bash
+aws ec2 run-instances \
+    --image-id ami-088ff0e3bde7b3fdf \
+    --instance-type t2.micro \
+    --count 1 \
+    --subnet-id subnet-08bf2850 \
+    --key-name  JupmBoxKeyForFirstEC2Instance \
+    --security-group-ids sg-06ae9a4efdd9232b8 \
+    --associate-public-ip-address \
+    --tag-specifications 'ResourceType=instance,Tags=[{Key=Env,Value=DAtest}]' 
```

Comment:
Nothing wrong with a tag "Eng", but if you create a tag "Name" it will show that name on the console, which can be useful.